File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 1 # Where the version number matches the one in the test.
end

module Hamming
  extend self
  def compute(str1, str2)
    distance = 0
    str1.split('').each_with_index do |c, i|
      distance += 1 if c != str2[i] 
    end
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {}

Hamming.prototype.compute = function(str1, str2) {
	distance = 0;
	if(str1.length != str2.length) {
		throw Error("DNA strands must be of equal length.");
	}
	for(i=0; i < str1.length; i++) {
		if(str1[i] != str2[i]) {
			distance++;
		}
	}
	return distance;
}
module.exports = Hamming;
